presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -630,6 +631,21 @@ private void addStatsListener(QueryExecution queryExecution)
     {
         Object lock = new Object();
 
+        AtomicBoolean queued = new AtomicBoolean();
+        queryExecution.addStateChangeListener(newValue -> {
```

Comment:
I did a test on a private build and the change listener was called for Queued. I was thinking of doing something simpler, just call queryQueued() before the call to addStatsListener.